tests/events/events.go

```diff
@@ -0,0 +1,105 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type k8sObject interface {
+	metav1.Object
+	schema.ObjectKind
+}
+
+// ExpectNoEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectNoEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be not triggered")
+	expectEvent(object, eventType, reason, BeEmpty())
+}
+
+// ExpectEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectEvent(object k8sObject, eventType, reason string) {
+	By("Expecting for an event to be triggered")
+	expectEvent(object, eventType, reason, Not(BeEmpty()))
+}
+
+// DeleteEvents is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func DeleteEvents(object k8sObject, eventType, reason string) {
+	By("Expecting events to be removed")
+	virtClient, err := kubecli.GetKubevirtClient()
+	util.PanicOnError(err)
+
+	fieldSelector, namespace := constructFieldSelectorAndNamespace(object, eventType, reason)
+
+	events, err := virtClient.CoreV1().Events(namespace).List(
+		context.Background(),
+		metav1.ListOptions{
+			FieldSelector: fieldSelector,
+		},
+	)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	for _, event := range events.Items {
+		err = virtClient.CoreV1().Events(event.Namespace).Delete(context.Background(), event.Name, metav1.DeleteOptions{})
```

Comment:
Instead of `List` + for loop, can we use `DeleteCollection` with the proper `ListOptions`?